src/storage/file_io.c

```diff
@@ -7631,60 +7596,52 @@ fileio_delete_queue_head (FILEIO_QUEUE * queue_p)
 static int
 fileio_compress_backup_node (FILEIO_NODE * node_p, FILEIO_BACKUP_HEADER * backup_header_p)
 {
-  int error = NO_ERROR;
-  int rv;
+  int error = NO_ERROR, local_buf_len;
+  FILEIO_ZIP_PAGE *zip_page;
 
-  if (!node_p || !backup_header_p)
+  if (!node_p || !node_p->zip_info || !backup_header_p)
     {
       goto exit_on_error;
     }
 
   assert (node_p->nread >= 0);
 
+  zip_page = &node_p->zip_info->zip_page;
```

Comment:
it looks need to check if zip_page is null.

Reply:
related pointers that can be null are checked: node_p & node_p->zip_info. `zip_page = &node_p->zip_info->zip_page` is impossible to be null (FILEIO_ZIP_PAGE is contained in the zip_info struct) therefore I don't see a reason to do the check.

Reply:
Ok, I understood.